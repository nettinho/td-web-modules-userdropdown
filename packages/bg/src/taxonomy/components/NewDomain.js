import _ from "lodash/fp";
import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Container, Header, Icon, Segment } from "semantic-ui-react";
import { FormattedMessage } from "react-intl";
import { createDomain } from "../routines";
import DomainForm from "./DomainForm";
import DomainCrumbs from "./DomainCrumbs";

export const NewDomain = ({ domain, createDomain }) => (
  <Fragment>
    <DomainCrumbs actionCrumb="domains.actions.create" />
    <Container as={Segment} text>
      <Header as="h2">
        <Icon name="cube" />
        <Header.Content>
          {!_.isEmpty(domain) && (
            <FormattedMessage id="domain.actions.create" />
          )}
          {_.isEmpty(domain) && (
            <FormattedMessage id="domains.actions.create" />
          )}
        </Header.Content>
      </Header>
      <DomainForm
        handleSubmit={createDomain}
        domain={{ parent_id: domain.id }}
      />
    </Container>
  </Fragment>
);

NewDomain.propTypes = {
  domain: PropTypes.object.isRequired,
  createDomain: PropTypes.func.isRequired
};

const mapStateToProps = ({ domain }) => ({ domain });
const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ createDomain }, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NewDomain);
