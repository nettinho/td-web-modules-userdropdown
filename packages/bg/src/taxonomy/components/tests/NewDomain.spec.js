import React from "react";
import { shallow } from "enzyme";
import { NewDomain } from "../NewDomain";

describe("<NewDomain />", () => {
  const createDomain = jest.fn();
  const domain = { id: 1, name: "nn", description: "dd" };
  const props = { domain, createDomain };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<NewDomain {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
