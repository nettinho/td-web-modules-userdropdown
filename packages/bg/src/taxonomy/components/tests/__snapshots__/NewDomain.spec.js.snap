// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewDomain /> matches the latest snapshot 1`] = `
<React.Fragment>
  <InjectIntl(Connect(DomainCrumbs))
    actionCrumb="domains.actions.create"
  />
  <Container
    as={[Function]}
    text={true}
  >
    <Header
      as="h2"
    >
      <Icon
        as="i"
        name="cube"
      />
      <HeaderContent>
        <FormattedMessage
          id="domain.actions.create"
          values={Object {}}
        />
      </HeaderContent>
    </Header>
    <withRouter(InjectIntl(Connect(DomainForm)))
      domain={
        Object {
          "parent_id": 1,
        }
      }
      handleSubmit={[MockFunction]}
    />
  </Container>
</React.Fragment>
`;
